helpers/target.go

```diff
@@ -129,3 +154,332 @@ func isAllowed(addr string) (bool, error) {
+// to perform an assume role request.
+type AWSCreds struct {
+	AssumeRoleURL string
+	Role          string
+}
+
+// NewAWSCreds creates a new AWS Credentials for Assume Role.
+func NewAWSCreds(assumeRoleURL, role string) *AWSCreds {
+	return &AWSCreds{
+		AssumeRoleURL: assumeRoleURL,
+		Role:          role,
+	}
+}
+func (c *AWSCreds) URL() string {
+	return c.AssumeRoleURL
+}
+func (c *AWSCreds) Username() string {
+	return c.Role
+}
+func (c *AWSCreds) Password() string {
+	return ""
+}
+
+type DockerCreds struct {
+	RegistryURL string
+	User        string
+	Pass        string
+}
+
+// DockerHubCreds represents a void
+// DockerCreds struct allowed to be
+// used with Docker Hub registry.
+var DockerHubCreds = &DockerCreds{}
+
+// NewDockerCreds creates a new Docker Credentials struct.
+func NewDockerCreds(registryURL, user, pass string) *DockerCreds {
+	return &DockerCreds{
+		RegistryURL: registryURL,
+		User:        user,
+		Pass:        pass,
+	}
+}
+func (c *DockerCreds) URL() string {
+	return c.RegistryURL
+}
+func (c *DockerCreds) Username() string {
+	return c.User
+}
+func (c *DockerCreds) Password() string {
+	return c.Pass
+}
+
+type GitCreds struct {
+	RepoURL string
+	User    string
+	Pass    string
+}
+
+// NewGitCreds creates a new Git Credentials struct.
+// user and pass can be void if no auth is required.
+func NewGitCreds(repoURL, user, pass string) *GitCreds {
+	return &GitCreds{
+		RepoURL: repoURL,
+		User:    user,
+		Pass:    pass,
+	}
+}
+func (c *GitCreds) URL() string {
+	return c.RepoURL
+}
+func (c *GitCreds) Username() string {
+	return c.User
+}
+func (c *GitCreds) Password() string {
+	return c.Pass
+}
+
+// IsReachable returns wether target is reachable
+// so the check execution can be performed.
+// If asset type is void, it will be inferred from target.
+//
+// ServiceCredentials are required for AWS, Docker and Git types.
+// Constructors for AWS, Docker and Git credentials can be found
+// in this same package.
+//
+// Verifications made depend on the asset type:
+//    - IP: None.
+//    - IPRange: None.
+//    - DomainName: None.
+//    - Hostname: NS Lookup resolution.
+//    - WebAddress: HTTP GET request.
+//    - AWSAccount: Assume Role.
+//    - DockerImage: Docker pull.
+//    - GitRepository: Git clone.
+//
+// This function does not return any output related to the process in order to
+// verify the target's reachability. This output can be useful for some cases
+// in order to not repeat work in the check execution (e.g.: Obtaining the
+// Assume Role token). For this purpose other individual methods can be called
+// from this same package with further options for AWS, Docker and Git types.
+func IsReachable(target, assetType string, creds ServiceCreds) (bool, error) {
+	var isReachable bool
+	var err error
+
+	// In order to support backward compatibility
+	// we have to support assetType being void.
+	if assetType == "" {
+		assetType, err = detectAssetType(target)
```

Comment:
I'm unsure about moving forward with this method and marking the check as inconclusive when the asset type hasn't been provided. Let's discuss this further.

Reply:
The problem is that we will loose backward compatibility for cases when the asset type is not provided, and as of now, if I'm not wrong, we should support this.

Reply:
Reachability checking support for backward compatible cases on which asset type was not set has been removed, now if assetType is not set explicitly, reachabilty tests will be skipped returning `true`. See comments below regarding backward compatibility.